logsmanagement/logsmanagement.c

```diff
@@ -0,0 +1,637 @@
+            info("Buffer flush to DB for %s out of range. Using minimum permitted value: %d", config_section->name, buff_flush_to_db_interval);
+        } 
+        
+        info("Buffers flush to DB interval (in sec) for %s will be set to: %d.\n", config_section->name, buff_flush_to_db_interval);
+
+        int64_t blob_max_size =  (appconfig_get_number(  &log_management_config, 
+                                                        config_section->name, 
+                                                        "disk space limit", 500) MiB) / BLOB_MAX_FILES;
+        
+        int update_every = appconfig_get_number(&log_management_config, config_section->name, 
+                                                "update every", g_logs_manag_update_every);
+        if(update_every < g_logs_manag_update_every) update_every = g_logs_manag_update_every;
+        info("Update every for %s: %d", config_section->name, update_every);
+
+        /* Check if log source path exists and is valid */  
+        char *log_path = appconfig_get(&log_management_config, config_section->name, "log path", NULL);
+        info("Log path (for %s):%s\n", config_section->name, log_path ? log_path : "NULL!");
+        if(!log_path || !*log_path || !strcmp(log_path, "auto") || access(log_path, F_OK)){ 
+            freez(log_path);
+            switch(log_type){
+                case FLB_SYSTEMD:
+                    log_path = strdupz(SYSTEMD_DEFAULT_PATH);
+                    break;
+                case FLB_DOCKER_EV:
+                    log_path = strdupz(DOCKER_EV_DEFAULT_PATH);
+                    break;
+                default:
+                    error("%s type requires a path.", log_source_t_str[log_type]);
+                    goto next_section;
+            }
+        } 
+
+        /* Check if log monitoring initialisation is successful */
+        // TODO: Add option to enable parser only without log storage and queries??
+        struct File_info *p_file_info = monitor_log_file_init( log_path, log_type, circular_buffer_max_size, 
+                                                               compression_accel, buff_flush_to_db_interval,
+                                                               blob_max_size, update_every);
+        if(p_file_info) info("Monitoring for %s initialized successfully.", config_section->name);
+        else {
+            error("Monitoring initialization for %s failed.", config_section->name);
+            goto next_section; // monitor_log_file_init() was unsuccessful
+        }
+
+        /* Initialise chart name */
+        p_file_info->chart_name = strdupz(config_section->name);
+
+        /* Configure (optional) custom charts */
+        p_file_info->parser_cus_config = callocz(1, sizeof(Log_parser_cus_config_t *));
+        p_file_info->parser_metrics->parser_cus = callocz(1, sizeof(Log_parser_cus_metrics_t *));
+        for(int cus_off = 1;; cus_off++){
+
+            /* Read chart name config */
```

Comment:
The magic numbers 19, 24, and 5 below could also have a `define` for them. :)

Reply:
I don't think it matters that much since they are very localised variables, however I've changed it to use dynamic memory allocation for some cleaner code :)